src/content/release/breaking-changes/popscope-with-result.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Generic types in PopScope
+description: >
+  Added a generic type in PopScope class and updated the onPopInvoked
+  function signature.
+---
+
+## Summary
+
+Added a generic type in [`PopScope`][] class and updated the [`onPopInvoked`][]
+function signature to take in an additional result as an position parameter.
+
+## Context
+
+Previously, PopScope hadn't haven a way to access the pop result when `onPopInvoked`
+is called. To add support for accessing pop result, the generic type is added to
+PopScope class so that the `onPopInvoked` can access the type-safe result.
+
+## Description of change
+
+Added a generic type in `PopScope` class and updated the `onPopInvoked`
+function signature to take in an additional result as an position parameter.
+
+## Migration guide
+
+Code before migration
+
+```dart
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(
+    MaterialApp(
+      navigatorKey: nav,
+      home: PopScope(
+        canPop: false,
+        onPopInvoked: (bool didPop) {
+          if (didPop) {
+            return;
+          }
+          launchConfirmationDialog();
+        },
+        child: MyWidget(),
+      ),
+    ),
+  );
+}
+```
+
+Code after migration
+```dart
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(
+    MaterialApp(
+      navigatorKey: nav,
+      home: PopScope<Object?>(
+        canPop: false,
+        onPopInvoked: (bool didPop, Object? result) {
+          if (didPop) {
+            return;
+          }
+          launchConfirmationDialog();
+        },
+        child: MyWidget(),
+      ),
+    ),
+  );
+}
+```
+
+The generic type should match the generic type of the [`Route`] the `PopScope` is in.
+If the route uses `int` as its generic type, consider use `PopScope<int>`.
+
+If the widget are shared across multiple routes with different types, one can
+use `PopScope<Object?>` to catch all possible types.
+
+## Timeline
+
+Landed in version: TBD
```

Comment:
will fill this out when pr merges. It would require internal migration, so may take some time